### PR TITLE
Fix #3734: Add Brave services key.

### LINKED ISSUE #3734: Internal issue 771
https://github.com/brave/internal/issues/771

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #3734 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -143,7 +143,7 @@ platform :ios do
       sdk: "iphoneos",
       clean: true,
       output_directory: "build",
-      xcargs: "-allowProvisioningUpdates BRAVE_API_KEY=\"#{ENV['BRAVE_STATS_API_KEY']}\""
+      xcargs: "-allowProvisioningUpdates BRAVE_API_KEY=\"#{ENV['BRAVE_STATS_API_KEY']}\" BRAVE_SERVICES_KEY=\"#{ENV['BRAVE_SERVICES_KEY']}\""
```
Comment (kylehickinson) [inline fastlane/Fastfile:146]: Do we anticipate the key being different across different environments (staging/dev/prod)?

**Review by kylehickinson — APPROVED** [review]

**Review by Brandon-T — APPROVED** [review]

File: fastlane/Fastfile (hunk 1 above)
Comment (iccub) [reply fastlane/Fastfile:146]: question for @jumde, how did you solve it for desktop? you inject it at CI level so it's different

Comment (iccub) [reply fastlane/Fastfile:146]: And is this good enough for prod now, as we want to push to prod as soon as possible

Comment (jumde) [reply fastlane/Fastfile:146]: @kylehickinson - the key will be the same for all environments.

**Review by jumde — APPROVED** [review]
lgtm

**Pull request merged**